ic-agent/src/agent/mod.rs

```diff
@@ -242,6 +243,11 @@ pub struct Agent {
     transport: Arc<dyn ReplicaV2Transport>,
 }
 
+#[derive(CandidType, Deserialize)]
+struct CreateCanisterResult {
+  canister_id: candid::Principal,
```

Comment:
Why cannot I change `candid::Principal` to `Principal` here? I could change it if I replace `let paths = cert.tree.list_paths();` by `let paths: Vec<Vec<Label> > = vec![];` on l. 631--632 and drop `git = ...` for the `ic-types` dependency in `Cargo.toml`.

Reply:
Resolved thanks to @Daniel-Bloom-dfinity